packages/jaeger-ui/src/model/ddg/types.tsx

```diff
@@ -53,3 +55,56 @@ export type TDdgModel = {
 };
 
 export type TDdgVertex = TVertex<{ service: string; operation: string }>;
+
+export type TDdgStateEntry =
+  | {
+      state: typeof fetchedState.LOADING;
+    }
+  | {
+      error: ApiError;
+      state: typeof fetchedState.ERROR;
+    }
+  | {
+      model: TDdgModel;
+      state: typeof fetchedState.DONE;
+      viewModifiers: Map<number, number>;
+    };
+
+export const stateKey = ({ service, operation = '*', start, end }: TDdgModelParams): string =>
+  [service, operation, start, end].join('\t');
+
+export type TDdgState = Record<string, TDdgStateEntry>;
+
+export enum EViewModifier {
+  None,
+  Hovered,
+  Selected,
+  Emphasized = 1 << 2, // eslint-disable-line no-bitwise
+}
+
+export type TDdgModelParams = {
+  service: string;
+  operation?: string;
+  start: number;
+  end: number;
+};
+
+export type TDdgActionMeta = {
+  query: TDdgModelParams;
+};
+
+export type TDdgAddViewModifierPayload = TDdgModelParams & {
+  visibilityIndices: number[];
+  viewModifier: number;
```

Comment:
Why not use `EViewModifier`, instead of `number`, as the type, here?

Reply:
Added a comment for this:
```
+  // Number instead of EViewModifier so that multiple views can be changed at once.
```